code/game/objects/structures/plasticflaps.dm

```diff
@@ -0,0 +1,64 @@
+/obj/structure/plasticflaps //HOW DO YOU CALL THOSE THINGS ANYWAY
+	name = "plastic flaps"
+	desc = "Completely impassable - or are they?"
+	icon = 'icons/obj/stationobjs.dmi' //Change this.
+	icon_state = "plasticflaps"
+	density = 0
+	anchored = 1
+	layer = 4
+	explosion_resistance = 5
+	var/list/mobs_can_pass = list(
+		/mob/living/carbon/slime,
+		/mob/living/simple_animal/mouse,
+		/mob/living/silicon/robot/drone
+		)
+
+/obj/structure/plasticflaps/CanPass(atom/A, turf/T)
+	if(istype(A) && A.checkpass(PASSGLASS))
+		return prob(60)
+
+	var/obj/structure/bed/B = A
+	if (istype(A, /obj/structure/bed) && B.buckled_mob)//if it's a bed/chair and someone is buckled, it will not pass
+		return 0
+
+	if(istype(A, /obj/vehicle))	//no vehicles
+		return 0
+
+	var/mob/living/M = A
+	if(istype(M))
+		if(M.lying)
+			return ..()
+		for(var/mob_type in mobs_can_pass)
```

Comment:
`is_type_in_list()`?

Reply:
I moved a bunch of code around, wasn't planning on refactoring any of it.

Reply:
Fair enough.